src/wallet/wallet.cpp

```diff
@@ -1391,7 +1398,7 @@ void CWallet::SyncTransaction(const CTransactionRef& ptx, CWalletTx::Status stat
 void CWallet::TransactionAddedToMempool(const CTransactionRef& ptx) {
     auto locked_chain = chain().lock();
     LOCK(cs_wallet);
-    SyncTransaction(ptx, CWalletTx::Status::UNCONFIRMED, {} /* block hash */, 0 /* position in block */);
+    SyncTransaction(ptx, CWalletTx::Status::UNCONFIRMED, {} /* block hash */, 0 /* position in block */, 0 /* block height */);
```

Comment:
In commit "Pass block height in Chain::BlockConnected/Chain::BlockDisconnected" (82c591ea35ac918147ca66444ae608b8385abd7c)

Order of arguments here is wrong. Block height come before position in block.

Reply:
Added commit d0c4ce6 : Replace CWalletTx:SetConf by Confirmation initialization list, should make things less error-prone.